src/network/networkd-dhcp-common.c

```diff
@@ -505,12 +544,34 @@ int config_parse_section_route_table(
                 return 0;
         }
 
-        if (STRPTR_IN_SET(section, "DHCP", "DHCPv4")) {
+        switch(ltype) {
+        case AF_INET:
                 network->dhcp_route_table = rt;
                 network->dhcp_route_table_set = true;
-        } else { /* section is IPv6AcceptRA */
+                network->dhcp_route_table_set_explicitly = true;
+                break;
+        case AF_INET6+RTPROT_DHCP:
```

Comment:
hmm, so AF_INET6 is 10, RTPROT_DHCP is 16, so this is 26. But 26 is also AF_LLC...

Instead of simple addition (which results in values the AF_XYZ constants are in anyway), maybe do `RTPROT_DHCP << 16 | AF_INET6` or so?

That should make collisions less likely, but be equally easy to work with?